bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,901 @@
+        try {
+            ContentResponse response = httpClient.GET(BASEURL + url + "?_=" + System.currentTimeMillis());
+            String pattern = "(?m)^\\s*\\r?\\n|\\r?\\n\\s*(?!.*\\r?\\n)";
+            String replacement = "";
+            String content = response.getContentAsString();
+            logger.debug("HTTP Response ({}) Body:{}", response.getStatus(), content.replaceAll(pattern, replacement));
+
+            if (response.getStatus() == HttpStatus.OK_200) {
+                result = gson.fromJson(content, jsonClass);
+            }
+            return result;
+        } catch (ExecutionException e) {
+            logger.warn("Caught ExecutionException {} for URL string {}", e, url);
+        } catch (InterruptedException e) {
+            logger.warn("Caught InterruptedException {} for URL string {}", e, url);
+        } catch (TimeoutException e) {
+            logger.warn("Caught TimeoutException {} for URL string {}", e, url);
+        }
+        return null;
+    }
+
+    private @Nullable ContentResponse postVerisureAPI(String url, String data, Boolean isJSON) {
+        try {
+            logger.debug("postVerisureAPI URL: {} Data:{}", url, data);
+            Request request = httpClient.newRequest(url).method(HttpMethod.POST);
+            if (isJSON) {
+                request.header("content-type", "application/json");
+            } else {
+                if (csrf != null) {
+                    request.header("X-CSRF-TOKEN", csrf);
+                }
+            }
+            request.header("Accept", "application/json");
+            if (!data.equals("empty")) {
+                request.content(new BytesContentProvider(data.getBytes(StandardCharsets.UTF_8)),
+                        "application/x-www-form-urlencoded; charset=UTF-8");
+            } else {
+                logger.debug("Setting cookie with username {} and vid {}", userName, password);
+                request.cookie(new HttpCookie("username", userName));
+                request.cookie(new HttpCookie("vid", password));
+            }
+            logger.debug("HTTP POST Request {}.", request.toString());
+            return request.send();
+        } catch (ExecutionException e) {
+            logger.warn("Caught ExecutionException {}", e.getMessage(), e);
+        } catch (InterruptedException e) {
+            logger.warn("Caught InterruptedException {}", e.getMessage(), e);
+        } catch (TimeoutException e) {
+            logger.warn("Caught TimeoutException {}", e.getMessage(), e);
+        } catch (RuntimeException e) {
+            logger.warn("Caught RuntimeException {}", e.getMessage(), e);
+        }
+        return null;
+    }
+
+    private @Nullable <T> T postJSONVerisureAPI(String url, String data, Class<T> jsonClass) {
+        for (int cnt = 0; cnt < APISERVERLIST.size(); cnt++) {
+            ContentResponse response = postVerisureAPI(apiServerInUse + url, data, Boolean.TRUE);
```

Comment:
This code looks like it is trying to iterate through APISERVERLIST but it doesn't iterate through them directly. Instead it uses the `apiServerInUse` field and `getNextApiServer()` to move through the list. This make the code highly unintuitive and extremely prone to developer mistakes. Please change this code to a simple iteration with a local variable, don't use a class field like `apiServerInUse`. Please remove the `apiServerInUse` field and `getNextApiServer()` method. Instead of `apiServerInUse` pass the server as a parameter in method calls.

Reply:
I agree that the code looks confusing, but I need to know which APISERVER that is in use since that can last for weeks, if I would just iterate the list I would have the chance to chose wrong APISERVER for every POST that gives a JSON.

Do you see my point?

Reply:
@cpmeister Did you see my remark regarding this comment?

Reply:
I saw your comment but didn't want to respond until I could recommend a viable alternative.
Are these other api servers simply fallback api servers in case the other fail? Or do the api servers host different parts of the api? Just trying to get an idea of your use case.

Reply:
They are fallback API servers that seems to be in use for a couple of days or weeks until they might switch to the other one and the use that for som arbitrary time before switching back again.